src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewDataDictionaryControllerPropertyActivator.cs

```diff
@@ -14,24 +13,30 @@ namespace Microsoft.AspNetCore.Mvc.ViewFeatures
     public class ViewDataDictionaryControllerPropertyActivator : IControllerPropertyActivator
     {
         private readonly Func<Type, PropertyActivator<ControllerContext>[]> _getPropertiesToActivate;
+        private readonly ControllerViewDataDictionaryFactory _viewDataDictionaryFactory;
         private readonly IModelMetadataProvider _modelMetadataProvider;
-        private ConcurrentDictionary<Type, PropertyActivator<ControllerContext>[]> _activateActions;
-        private bool _initialized;
-        private object _initializeLock = new object();
+        private readonly ConcurrentDictionary<Type, PropertyActivator<ControllerContext>[]> _activateActions
+            = new ConcurrentDictionary<Type, PropertyActivator<ControllerContext>[]>();
 
+        [Obsolete("This constructor is obsolete and will be removed in a future version")]
+        // Note: Remove Legacy code path from GetViewDataDictionary whence this constructor is removed.
```

Comment:
Do you mean "when this constructor" or "from where this constructor"?

Reply:
I was trying to be fancy. "when" is the right term to use here.

Reply:
Yeah, whence means "from what place or source", but it's archaic so it's best to avoid it in code.